### PR TITLE
Implement IsIn on Column

### LINKED ISSUE #118: [FEATURE REQUEST]: IsIn
IsIn operator is not available on column as is available on the Scala API. Is it possible to add it?

### PR DESCRIPTION
**Pull request by GoEddie**
We are excited to review your PR.

So we can do the best job, please check:

- [x] There's a descriptive title that will make sense to other developers some time from now. 
- [x] There's associated issues. All PR's should have issue(s) associated - unless a trivial self-evident change such as fixing a typo. You can use the format `Fixes #nnnn` in your description to cause GitHub to automatically close the issue(s) when your PR is merged.

Fixes #118 

- [x] Your change description explains what the change does, why you chose your approach, and anything else that reviewers should know.

Adds the IsIn method to Column that takes a list of (specific types) and calls the isin in spark. The issue mention about supporting Column[] as an input but when I tested this in raw spark altough it takes an Any[] Column[] doesn't do anything so I stuck to the base types.

- [x] You have included any necessary tests in the same PR.

TestIsIn() in ColumnTests

### CODE REVIEW

File: src/csharp/Microsoft.Spark/Sql/Column.cs [hunk 1]
```diff
@@ -743,6 +743,80 @@ public Column Over()
             return Apply("over");
         }
 
+
+        /// <summary>
+        ///  A boolean expression that is evaluated to true if the value of this expression 
+        ///  is contained by the evaluated values of the arguments.  
+        /// </summary>
```
Comment (suhsteve) [inline src/csharp/Microsoft.Spark/Sql/Column.cs]: /// <param name="list">...</param>
Also add this to the others as well.

File: src/csharp/Microsoft.Spark.UnitTest/Sql/ColumnTests.cs [hunk 2]
```diff
@@ -363,6 +363,28 @@ public void TestOver()
...
+                var list = new System.Collections.Generic.List<int>(){0, 1, 99};
+                column1.IsIn(list);            
+                VerifyNonStaticCall(column1, "isin", list);
+            }
+                        {
+                Column column1 = CreateColumn("col1");
+                var list = new System.Collections.Generic.List<float>(){0F, 1F, 99F};
+                column1.IsIn(list);            
+                VerifyNonStaticCall(column1, "isin", list);
+            }
```
Comment (suhsteve) [inline src/csharp/Microsoft.Spark.UnitTest/Sql/ColumnTests.cs]: can you test the other `IsIn(..)` method signatures as well ?

File: src/csharp/Microsoft.Spark/Sql/Column.cs [hunk 3]
```diff
@@ -743,6 +743,80 @@ public Column Over()
...
+        /// <returns>Column object</returns>
+        public Column IsIn(IEnumerable<string> list){
```
Comment (imback82) [inline src/csharp/Microsoft.Spark/Sql/Column.cs]: The Scala version takes in varargs, so we should do something similar with `params`.

**Comment by GoEddie** [comment]
Hi,

Do you need anything else on this or is the last update good enough?

Thanks,


Ed

File: src/csharp/Microsoft.Spark.UnitTest/Sql/ColumnTests.cs [hunk 4]
```diff
@@ -363,6 +363,115 @@ public void TestOver()
             }
         }
 
+        [Fact]
+        public void TestIsIn(){
```
Comment (imback82) [inline src/csharp/Microsoft.Spark.UnitTest/Sql/ColumnTests.cs]: Can you please follow the existing coding style?

File: src/csharp/Microsoft.Spark/Sql/Column.cs [hunk 5]
```diff
@@ -743,6 +743,167 @@ public Column Over()
             return Apply("over");
         }
 
+        /// <summary>
+        ///  A boolean expression that is evaluated to true if the value of this expression 
+        ///  is contained by the evaluated values of the arguments.  
+        /// </summary>
+        /// <param name="list">List of values to check the column against</param>
+        /// <returns>Column object</returns>
+        public Column IsIn(params string[] list){
```
Comment (imback82) [inline src/csharp/Microsoft.Spark/Sql/Column.cs]: Please follow the existing coding style.

File: src/csharp/Microsoft.Spark/Sql/Column.cs [hunk 6]
```diff
@@ -743,6 +743,167 @@ public Column Over()
...
+            return Apply("isin", list);
+        }      
+                
+        /// <summary>
+        ///  A boolean expression that is evaluated to true if the value of this expression 
+        ///  is contained by the evaluated values of the arguments.  
+        /// </summary>
+        /// <param name="list">List of values to check the column against</param>
+        /// <returns>Column object</returns>
+        public Column IsIn(List<string> list){
```
Comment (imback82) [inline src/csharp/Microsoft.Spark/Sql/Column.cs]: Why do we expose this API when we already have params. Same for all the APIs below.

File: src/csharp/Microsoft.Spark.UnitTest/Sql/ColumnTests.cs (hunk 4 above)
Comment (GoEddie) [reply src/csharp/Microsoft.Spark.UnitTest/Sql/ColumnTests.cs]: i have added the indentation and the newline before the brace

File: src/csharp/Microsoft.Spark/Sql/Column.cs (hunk 6 above)
Comment (GoEddie) [reply src/csharp/Microsoft.Spark/Sql/Column.cs]: orginally it only had the List parameters but someone asked to add in the params versions - i have taken the list ones out now

File: src/csharp/Microsoft.Spark/Sql/Column.cs (hunk 5 above)
Comment (GoEddie) [reply src/csharp/Microsoft.Spark/Sql/Column.cs]: newline now before braces - sorry didn't see it for the life of me in the original functions!

File: src/csharp/Microsoft.Spark.UnitTest/Sql/ColumnTests.cs [hunk 7]
```diff
@@ -363,6 +363,68 @@ public void TestOver()
             }
         }
 
+        [Fact]
+        public void TestIsIn()
+        {   
+            {
+                var expected = new System.Collections.Generic.List<string> {"vararg_1", "vararg_2"};
```
Comment (imback82) [inline src/csharp/Microsoft.Spark.UnitTest/Sql/ColumnTests.cs]: Can we use `using` for `System.Collections.Generic.List`?

File: src/csharp/Microsoft.Spark.UnitTest/Sql/ColumnTests.cs [hunk 8]
```diff
@@ -363,6 +363,68 @@ public void TestOver()
...
+                Column column1 = CreateColumn("col1");
```
Comment (imback82) [inline src/csharp/Microsoft.Spark.UnitTest/Sql/ColumnTests.cs]: You can rename all `column1` to `column`.

File: src/csharp/Microsoft.Spark.UnitTest/Sql/ColumnTests.cs (hunk 7 above)
Comment (GoEddie) [reply src/csharp/Microsoft.Spark.UnitTest/Sql/ColumnTests.cs]: done

File: src/csharp/Microsoft.Spark.UnitTest/Sql/ColumnTests.cs (hunk 8 above)
Comment (GoEddie) [reply src/csharp/Microsoft.Spark.UnitTest/Sql/ColumnTests.cs]: I just wanted to check you did want this as the existing tests used the same Column column1 = CreateColumn("col1") - even when the tests just create one column.

Let me know if you do want me to rename it in the IsIn tests as well :)

Comment (imback82) [reply src/csharp/Microsoft.Spark.UnitTest/Sql/ColumnTests.cs]: Ok it is already existing tests, then it's fine. I will clean up later. Thanks!

Comment (GoEddie) [reply src/csharp/Microsoft.Spark.UnitTest/Sql/ColumnTests.cs]: great - if you raise an issue and say what you want (if one column no number, what if two columns or more, start at 1 then) am happy to do it as a separate pr

**Pull request merged**